src/mechanics/tools/QuasistaticSolver.cpp

```diff
@@ -118,6 +123,21 @@ GlobalDofVector QuasistaticSolver::ToGlobalDofVector(const Eigen::VectorXd& x)
     return v;
 }
 
+void QuasistaticSolver::WriteTimeDofResidual(std::ostream& out, DofType dofType, std::vector<int> dofNumbers)
+{
+    mGlobalTime -= mTimeStep;
```

Comment:
Maybe you could add a short comment as to why you need to subtract a time step...